src/config.js

```diff
@@ -50,7 +50,7 @@ export function applyConfigToArgv({
         `an unknown option: "${option}"`);
     }
     if (options[decamelizedOptName].type === undefined) {
-      throw new WebExtError(
+      throw new UsageError(
```

Comment:
@kumar303 ah, you are definitely right, and this change is my fault, it is related to this comment https://github.com/mozilla/web-ext/pull/1080#discussion_r141647124 (Sorry @saintsebastian).

Nevertheless, to prevent this check from be read wrong in the future, how about adding an inline comment near this `throw` to briefly explain what this error is about?

Another interesting option (I can file a bug so that we can handle it in a follow up) would be to use a more specific type in the flow annotations here:

https://github.com/mozilla/web-ext/blob/a55ee4c9ef59c590f0016be7984ac7ca63b27ea0/src/program.js#L96

so that we can make flow to raise an error during its static analysis if any of the options defined in program.js is missing the type property.

Reply:
@rpl good idea about defining the shape of `options`

Reply:
Is this PR a good place to add that type? @kumar303 @rpl

Reply:
@kumar303 @saintsebastian we could definitely define and add the additional flow types for the options in this PR, the amount of changes should not be that big, but it could be reasonable to handle it in as a separate issue and pull requests, so that we are free to discuss the details of it without blocking this PR.

In the meantime, I added an issue related to it as #1144 (which also contains some additional info about the change, e.g. where these new flow types would be used and what their flow type definitions could probably look like).